src/main/java/uk/gov/laa/gpfd/config/JwtAuthenticationFilter.java

```
@@ -17,17 +18,76 @@
+
+    public JwtAuthenticationFilter(JwtDecoder jwtDecoder, AppConfig appConfig) {
+        this.jwtDecoder = jwtDecoder;
+        this.appConfig = appConfig;
+    }
 
     @Override
     public void doFilterInternal(HttpServletRequest servletRequest, @NotNull HttpServletResponse servletResponse, @NotNull FilterChain filterChain) throws IOException, ServletException {
         var token = servletRequest.getHeader("Authorization");
 
-        if (token != null && !token.isEmpty()) {
-            // JWT validation logic to come in later PR
+        if (token == null || token.isEmpty()) {
+            filterChain.doFilter(servletRequest, servletResponse);
+        } else {
+            validateJwt(token);
+            filterChain.doFilter(servletRequest, servletResponse);
         }
+    }
+
+    public boolean validateJwt(String token) {
+        var jwtContent = extractJwtToken(token);
+
+        try {
+            Jwt decodedToken = jwtDecoder.decode(jwtContent);
+
+            if (decodedToken == null)
+                throw new JwtException("decode token returned null");
+
+            String username = decodedToken.getSubject();
+
+            if (username == null || username.isEmpty())
+                throw new JwtException("token includes no valid username");
+
+            if (!decodedToken.getAudience().contains(appConfig.getEntraIdClientId())) {
+                throw new JwtException("Audience mismatch");
+            }
 
-        filterChain.doFilter(servletRequest, servletResponse);
+            if (!decodedToken.getClaimAsString(JWT_PAYLOAD_TENANT_ID_KEY).equals(appConfig.getEntraIdTenantId())) {
+                throw new JwtException("Incorrect Tenant ID");
+            }
+
+            if (!decodedToken.getClaimAsString(JWT_PAYLOAD_APPLICATION_ID_KEY).equals(appConfig.getEntraIdClientId())) {
+                throw new JwtException("Incorrect Application ID");
+            }
+
+            if (!isTokenValidForCurrentTime(decodedToken)) {
```

Currently, we check isTokenValidForCurrentTime() before isTokenExpired(). Since expired tokens should always be rejected immediately, we could potentially:
- Check expiry first - This would fail fast for the most common invalid case
- Then validate not-before time
- Finally verify other claims (audience, tenant ID, etc.)

---

Resolved as requested